lib/uksignal/include/ksigaction.h

```diff
@@ -0,0 +1,9 @@
+/* Imported from musl 1.2.3 */
```

Comment:
Should we add the licence header here?

Reply:
@andreittr, indeed its contents should be something like

```
/* SPDX-License-Identifier: BSD-3-Clause */
/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
 * Licensed under the BSD-3-Clause License (the "License").
 * You may not use this file except in compliance with the License.
 */

/* Imported from musl 1.2.3 */
/* In line with the Linux kernel struct for x86_64 and aarch64. */
#ifndef __KSIGACTION_H__
#define __KSIGACTION_H__

struct k_sigaction {
        void (*handler)(int);
        unsigned long flags;
        void (*restorer)(void);
        unsigned mask[2];
};

#endif /* __KSIGACTION_H__ */
```

@skuenzer, would this be OK with you?

Moreover, should this be **outside** the `include/` directory, since it's internal to `uksignal`, not exported? If it is OK to be in `include/`, then it should be in `include/uk/`, right?

Would it make sense to have the C++ guards?

```
#ifdef __cplusplus
extern "C" {
#endif

...

#ifdef __cplusplus
}
#endif 
``